### PR TITLE
fix(context): parsing zero as string

### PR DESCRIPTION
**Pull request by santiagocardo**
Fix context parsing with zero as a string.

### CODE REVIEW

File: lib/expression/context.ex [hunk 1]
```diff
@@ -50,11 +50,13 @@ defmodule Expression.Context do
...
 
   # Prevent implictly converting numbers starting with a zero
+  defp iterate({key, "0"}, _opts), do: {key, 0}
+
   defp iterate({key, "0" <> value}, opts) when is_binary(value) do
     if String.match?("0" <> value, ~r/^[0-9]/) do
       {key, "0" <> value}
     else
-      iterate({key, value}, opts)
+      iterate({key, "0" <> value}, opts)
```
Comment (fedme) [inline lib/expression/context.ex]: Isn't this calling itself recursively using the same exact arguments? I am confused

Comment (santiagocardo) [reply lib/expression/context.ex]: Yeah, nice catch. I'm fixing it now, thanks!
It went out as part of some debugging 🙃

Comment (santiagocardo) [reply lib/expression/context.ex]: Done in [8d52db4](https://github.com/turnhub/expression/pull/209/commits/8d52db470525e523c66560fcf7c6c9eeb34dd431)

**Review by fedme — APPROVED** [review]
Nice catch thanks!

**Review by smn — APPROVED** [review]

**Pull request merged**